planner/update.go

```diff
@@ -72,7 +72,15 @@ func (n *updateNode) Next() (bool, error) {
 			if err != nil {
 				return false, err
 			}
-			_, err = n.collection.UpdateWithDocID(n.p.ctx, docID, string(patch))
+			doc, err := n.collection.Get(n.p.ctx, docID, false)
```

Comment:
todo: Why are you fetching the value here? You already have it `n.currentValue = n.results.Value()`

Reply:
I haven't worked with the planner much. Is there a way to create a `client.Document` from a `core.Doc`?

Reply:
I missed that the types here were different, but yes you can - easiest way would probably be:
```
docMap := n.currentValue.ToMap()
clientDoc := client.NewDocFromMap(docMap, colDef)
```
There might be something nicer somewhere, but I couldn't spot anything in a 30sec scan.  I prefer the double-cast over fetching twice.

Reply:
Took a few tries but I think I found a good solution.

Reply:
It's actually not possible to update a document without fetching it first. Partial document updates break some parts of our indexing system.

It's also not possible to reconstruct the document from the planner node because it only selects the field values the user has requested be returned.

I've reverted back to fetching the document first and then updating with the patch fields.

Reply:
> Partial document updates break some parts of our indexing system.

Thats not good, and a regression IMO.  No worries though RE this PR, thanks for giving it a go :)

Reply:
@AndrewSisley That's exactly what `UpdateWithDocID` did so the pattern is the same. No regression in that sense.

Reply:
The regression was introduced with sec. indexes, I'm pretty sure that up until that point we could do partial updates via any public func, so long as the docID was provided.